programs/perpetuals/tests/anchor/test_client.ts

```diff
@@ -706,6 +708,92 @@ export class TestClient {
     }
   };
 
+  setCustomOraclePricePermissionless = async (
+    permissionlessOracleKeypair: Keypair,
+    price: number,
+    custody,
+    publishTime?,
+    noSignatureVerification?,
+    messageOverwrite?
+  ) => {
+    let user = this.users[0].wallet;
+
+    const schema = {
+      struct: {
+        custodyAccount: { array: { type: "u8", len: 32 } },
+        price: "u64",
+        expo: "i32",
+        conf: "u64",
+        ema: "u64",
+        publishTime: "i64",
+      },
+    };
+
+    let setCustomOraclePricePermissionlessParams = {
+      custodyAccount: custody.custody,
+      price: new BN(price * 1000),
+      expo: -3,
+      conf: new BN(0),
+      ema: new BN(price * 1000),
+      publishTime:
+        publishTime != null ? new BN(publishTime) : new BN(this.getTime()),
+    };
+
+    let message =
+      messageOverwrite != null
+        ? messageOverwrite
+        : borsh.serialize(schema, {
+            ...setCustomOraclePricePermissionlessParams,
+            custodyAccount:
+              setCustomOraclePricePermissionlessParams.custodyAccount.toBytes(),
+          });
+
+    const signature = nacl.sign.detached(
+      message,
+      permissionlessOracleKeypair.secretKey
+    );
+    let tx = new anchor.web3.Transaction();
+    if (noSignatureVerification == null) {
+      tx.add(
+        anchor.web3.Ed25519Program.createInstructionWithPublicKey({
+          publicKey: permissionlessOracleKeypair.publicKey.toBytes(),
+          message: message,
+          signature: signature,
+        })
+      );
+    }
+    tx.add(
+      this.program.instruction.setCustomOraclePricePermissionless(
+        setCustomOraclePricePermissionlessParams,
+        {
+          accounts: {
+            user: user.publicKey,
```

Comment:
you don't need to explicitly specify unused default signer as anchor signs with provider.wallet anyway

Reply:
that is good to know!!  removed `user` from the instruction and also dropped the signer here